### PR TITLE
Rename quay to docker in image bundles

### LINKED ISSUE #1233: Stop using "latest" tags on `kogito-images` and default the "development stream name"
Aligned with the branch name.

### PR DESCRIPTION
**Pull request by ricardozanini**
Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [ ] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [ ] Pull Request title is properly formatted: `[Issue XXXX] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

Fixes https://github.com/apache/incubator-kie-issues/issues/1233

@jstastny-cz I haven't changed the versions yet. We need to agree on the versions here. We have a version for the Kogito Platform, for the internal image modules, and finally the actual tag for the images. We can keep `999-SNAPSHOT` and handle it in CI. Or we can handle it in a script in this repo. Let me know the approach you prefer.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- <b>(Re)run Jenkins tests</b>  
  Please add comment: <b>Jenkins [test|retest] this</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by kie-ci3** [comment]
**PR job** `#11` was: **ABORTED**
Possible explanation: Most probably a timeout, please review

Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images.build-and-test/11/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
Starting building: KIE » kogito » main » pullrequest » kogito-images.build-image #52<br/>Calling Pipeline was cancelled<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image #55 completed: ABORTED<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-jobs-service-postgresql<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image #51 completed: ABORTED<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-jit-runner<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image #50 completed: ABORTED<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-jobs-service-allinone<br/>Click here to forcibly terminate running steps<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image PR #1768 - kogito-base-builder: https://github.com/apache/incubator-kie-kogito-images/pull/1768 completed: ABORTED<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-base-builder<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-data-index-postgresql<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-data-index-ephemeral<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-jobs-service-ephemeral<br/>[Pipeline] // parallel<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - 'https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images.build-and-test/11/api/json?depth=0'<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by kie-ci3** [comment]
**PR job** `#12` was: **ABORTED**
Possible explanation: Most probably a timeout, please review

Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images.build-and-test/12/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
Starting building: KIE » kogito » main » pullrequest » kogito-images.build-image #63<br/>Starting building: KIE » kogito » main » pullrequest » kogito-images.build-image #59<br/>Starting building: KIE » kogito » main » pullrequest » kogito-images.build-image #61<br/>Starting building: KIE » kogito » main » pullrequest » kogito-images.build-image #57<br/>Starting building: KIE » kogito » main » pullrequest » kogito-images.build-image #60<br/>Starting building: KIE » kogito » main » pullrequest » kogito-images.build-image #62<br/>Starting building: KIE » kogito » main » pullrequest » kogito-images.build-image #58<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image PR #1768 - kogito-base-builder: https://github.com/apache/incubator-kie-kogito-images/pull/1768 completed: SUCCESS<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image PR #1768 - kogito-jit-runner: https://github.com/apache/incubator-kie-kogito-images/pull/1768 completed: SUCCESS<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image PR #1768 - kogito-data-index-postgresql: https://github.com/apache/incubator-kie-kogito-images/pull/1768 completed: SUCCESS<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image PR #1768 - kogito-jobs-service-allinone: https://github.com/apache/incubator-kie-kogito-images/pull/1768 completed: SUCCESS<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-data-index-ephemeral<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-jobs-service-ephemeral<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-jobs-service-postgresql<br/>[Pipeline] // parallel<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - 'https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images.build-and-test/12/api/json?depth=0'<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by kie-ci3** [comment]
**PR job** `#14` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images.build-and-test/14/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
Starting building: KIE » kogito » main » pullrequest » kogito-images.build-image #67<br/>Starting building: KIE » kogito » main » pullrequest » kogito-images.build-image #68<br/>Starting building: KIE » kogito » main » pullrequest » kogito-images.build-image #66<br/>Starting building: KIE » kogito » main » pullrequest » kogito-images.build-image #69<br/>Starting building: KIE » kogito » main » pullrequest » kogito-images.build-image #65<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image PR #1768 - kogito-base-builder: https://github.com/apache/incubator-kie-kogito-images/pull/1768 completed: SUCCESS<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image PR #1768 - kogito-jobs-service-ephemeral: https://github.com/apache/incubator-kie-kogito-images/pull/1768 completed: SUCCESS<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image PR #1768 - kogito-jobs-service-postgresql: https://github.com/apache/incubator-kie-kogito-images/pull/1768 completed: SUCCESS<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image PR #1768 - kogito-data-index-postgresql: https://github.com/apache/incubator-kie-kogito-images/pull/1768 completed: SUCCESS<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image PR #1768 - kogito-jit-runner: https://github.com/apache/incubator-kie-kogito-images/pull/1768 completed: SUCCESS<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image PR #1768 - kogito-data-index-ephemeral: https://github.com/apache/incubator-kie-kogito-images/pull/1768 completed: SUCCESS<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Build KIE » kogito » main » pullrequest » kogito-images.build-image PR #1768 - kogito-jobs-service-allinone: https://github.com/apache/incubator-kie-kogito-images/pull/1768 completed: UNSTABLE<br/>[Pipeline] unstable<br/>WARNING: Tests on kogito-jobs-service-allinone seems to have failed<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>[Pipeline] // parallel<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - 'https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images.build-and-test/14/api/json?depth=0'<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

File: README.md [hunk 1]
```diff
@@ -166,12 +166,12 @@ The Kogito Component Images can be considered as lightweight images that will co
 by providing extra capabilities, like managing the processes on a web UI or providing persistence layer to the Kogito applications.
 Today we have the following Kogito Component Images:
 
-* [quay.io/kiegroup/kogito-data-index-ephemeral](https://quay.io/kiegroup/kogito-data-index-ephemeral)
-* [quay.io/kiegroup/kogito-data-index-postgresql](https://quay.io/kiegroup/kogito-data-index-postgresql)
-* [quay.io/kiegroup/kogito-jobs-service-ephemeral](https://quay.io/kiegroup/kogito-jobs-service-ephemeral) 
-* [quay.io/kiegroup/kogito-jobs-service-postgresql](https://quay.io/kiegroup/kogito-jobs-service-postgresql)
-* [quay.io/kiegroup/kogito-jobs-service-allinone](https://quay.io/kiegroup/kogito-jobs-service-allinone)
-* [quay.io/kiegroup/kogito-jit-runner](https://quay.io/kiegroup/kogito-jit-runner)
+* [docker.io/apache/incubator-kie-kogito-data-index-ephemeral](https://docker.io/apache/incubator-kie-kogito-data-index-ephemeral)
```
Comment (wmedvede) [inline README.md]: @ricardozanini General comment regarding this links for http navigation.

I'm not sure if we shouldn't have a link like this instead.  https://hub.docker.com/  bla bla bla.

see for example: https://hub.docker.com/_/postgres

Obviously not a blocker, we can adjust when we have the images publised and all the bits in place.

File: kogito-imagestream.yaml [hunk 2]
```diff
@@ -27,7 +27,7 @@ items:
   - kind: ImageStream
     apiVersion: v1
     metadata:
-      name: kogito-swf-builder
+      name: sonataflow-builder
```
Comment (wmedvede) [inline kogito-imagestream.yaml]: shouldn't this be kie-sonataflow-builder? 
If not, just ignore this comment.

**Review by wmedvede — APPROVED** [review]
LGTM, only minor comments from my side.

**Review by jstastny-cz — APPROVED** [review]
LGTM.

**Comment by jstastny-cz** [comment]
@ricardozanini I don't follow the problem with versions though - what is the difference between quay and dockerhub regarding our versioning?

**Comment by ricardozanini** [comment]
@jstastny-cz I think we will fix it in the way we see problems in the CI. But essentially this is related to now using `:main` tag instead of `-nightly:latest`.

**Comment by tiagobento** [comment]
...and `:10.0.x` for publications made from the 10.0.x branch.

**Comment by tiagobento** [comment]
So we can't really "hardcode" main there, because that would make the 10.0.x branch inconsistent.

**Comment by ricardozanini** [comment]
Once we branch, we must run a script to update this info in the repo.

**Comment by porcelli** [comment]
@ricardozanini are we good to merge this?

**Comment by ricardozanini** [comment]
> @ricardozanini are we good to merge this?

No, I'm working on a script to update the files in the command line.

**Comment by porcelli** [comment]
@ricardozanini ahh k, thank you for the update!

**Comment by ricardozanini** [comment]
@tiagobento we already have the script here: 

```shell
$ python manage-kogito-version.py --bump-to 10.0.x
```

I'm merging it. As soon as we define how to do the release/branch process we can adjust if necessary

**Comment by tiagobento** [comment]
@ricardozanini Will this update the kogito maven artifacts to 10.0.999-SNAPSHOT?

**Comment by ricardozanini** [comment]
Yes, it will fetch this version if available in Nexus.

**Pull request merged**